algorithms/src/Kokkos_Sort.hpp

```diff
@@ -456,18 +456,37 @@ class BinSort {
   void operator()(const bin_sort_bins_tag& /*tag*/, const int i) const {
     auto bin_size = bin_count_const(i);
     if (bin_size <= 1) return;
+    constexpr bool use_std_sort =
+#ifdef KOKKOS_ENABLE_SERIAL
+        std::is_same_v<exec_space, Kokkos::Serial> ||
+#endif
+#ifdef KOKKOS_ENABLE_OPENMP
+        std::is_same_v<exec_space, Kokkos::OpenMP> ||
+#endif
+        false;
     int lower_bound = bin_offsets(i);
     int upper_bound = lower_bound + bin_size;
-    for (int k = lower_bound + 1; k < upper_bound; ++k) {
-      int old_idx = sort_order(k);
-      int j       = k - 1;
-      while (j >= lower_bound) {
-        int new_idx = sort_order(j);
-        if (!bin_op(keys_rnd, old_idx, new_idx)) break;
-        sort_order(j + 1) = new_idx;
-        --j;
+    if (use_std_sort && bin_size > 10) {
```

Comment:
Where does the magic value "10" come from?

Reply:
I don't have a good motivation. The idea is that `std::sort` does insertion sort anyway for small values, so we save on the function call. But I can't tell what value is the best one.

Reply:
Please at a comment saying that 10 was found to be a good value experimentally.

Reply:
Quick check using ArborX:
```
for i in 0 1 3 5 10 15; do ./ArborX_$i --binary --filename ~/data/datasets/hacc_37M.arborx --algorithm mst --max-num-points 1000000 --core-min-size 1 2>/dev/null | grep Sorting | head -n 1; done
|   |   |       |   |-> 2.97e-02 sec 1.3% 98.8% 0.0% 0.0% 4.03e+02 1 ArborX::Sorting [region]
|   |   |       |   |-> 2.53e-02 sec 1.1% 98.4% 0.0% 0.0% 4.75e+02 1 ArborX::Sorting [region]
|   |   |       |   |-> 2.43e-02 sec 1.1% 98.5% 0.0% 0.0% 4.94e+02 1 ArborX::Sorting [region]
|   |   |       |   |-> 2.41e-02 sec 1.1% 98.6% 0.0% 0.0% 4.97e+02 1 ArborX::Sorting [region]
|   |   |       |   |-> 2.44e-02 sec 1.1% 98.6% 0.0% 0.0% 4.92e+02 1 ArborX::Sorting [region]
|   |   |       |   |-> 2.43e-02 sec 1.1% 98.6% 0.0% 0.0% 4.94e+02 1 ArborX::Sorting [region]
```
So, realistically, it's not very sensitive to it as long as it's at least 3. At least for the quick tries on a couple datasets. What comment would you want me to put in?

Reply:
```suggestion
    // Switching to std::sort for more than 10 elements was found to be reasonable experimentally.
    if (use_std_sort && bin_size > 10) {
```